tests/model/DataObjectTest.php

```diff
@@ -33,6 +33,25 @@ class DataObjectTest extends SapphireTest {
 		'ManyManyListTest_Category',
 	);
 
+	public function testSingleton() {
+		// Test that populateDefaults() isn't called on singletons, which can lead to SQL errors during build, and endless loops
+		$inst = DataObjectTest_Fixture::create();
```

Comment:
might be better to use a dataProvider here

Reply:
what would be the advantage?

Reply:
If one test fails it won't block all the other tests.

Reply:
Do the other tests in this class use dataProviders? Or can you provide an example? I'm not really sure how to proceed.

Reply:
@jonom see https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers

I've made the change for you and pushed it up

Reply:
> I've made the change for you and pushed it up

Thanks @dhensby! I didn't even realise you could do that. Were you able to push straight to my branch with your regular credentials or is it more complicated? I'll have to remember that option for the future.

Tests failed after your commit was introduced but [passed](https://travis-ci.org/silverstripe/silverstripe-framework/builds/339298668) before. I can't see anything wrong with your commit though so I'm quite confused.

Reply:
From the docs you linked:

> All data providers are executed before both the call to the setUpBeforeClass static method and the first call to the setUp method. Because of that you can't access any variables you create there from within a data provider. This is required in order for PHPUnit to be able to compute the total number of tests.

I guess SapphireTest:setUp() is running after the objects are instantiated?